### PR TITLE
Improve error message when trying to run a plan that does not exist from the CLI client

### LINKED ISSUE #299: Unclear error message when no plan exists
When trying to run `blueapi controller run non_existent_plan` I get:
```
Traceback (most recent call last):
  File "/scratch/ffv81422/testing_blueapi/.venv/bin/blueapi", line 8, in <module>
    sys.exit(main())
  File "/scratch/ffv81422/testing_blueapi/.venv/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/scratch/ffv81422/testing_blueapi/.venv/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/scratch/ffv81422/testing_blueapi/.venv/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/scratch/ffv81422/testing_blueapi/.venv/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/scratch/ffv81422/testing_blueapi/.venv/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/scratch/ffv81422/testing_blueapi/.venv/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/scratch/ffv81422/testing_blueapi/.venv/lib/python3.10/site-packages/blueapi/cli/cli.py", line 105, in wrapper
    func(*args, **kwargs)
  File "/scratch/ffv81422/testing_blueapi/.venv/lib/python3.10/site-packages/click/decorators.py", line 38, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/scratch/ffv81422/testing_blueapi/.venv/lib/python3.10/site-packages/blueapi/cli/cli.py", line 161, in run_plan
    resp = client.create_task(task)
  File "/scratch/ffv81422/testing_blueapi/.venv/lib/python3.10/site-packages/blueapi/cli/rest.py", line 68, in create_task
    return self._request_and_deserialize(
  File "/scratch/ffv81422/testing_blueapi/.venv/lib/python3.10/site-packages/blueapi/cli/rest.py", line 111, in _request_and_deserialize
    raise BlueskyRemoteError(str(response))
blueapi.cli.amq.BlueskyRemoteError: <Response [404]>
```

As a user this is not obvious to me that it's because the plan doesn't exist. Would be good to get an error message saying `non_existent_plan` doesn't exist and listing which plans do exist

### PR DESCRIPTION
**Pull request by DiamondJoseph**
- Closes #299

### CODE REVIEW

File: src/blueapi/cli/amq.py [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 from blueapi.worker import WorkerEvent
 
 
 class BlueskyRemoteError(Exception):
-    def __init__(self, message: str) -> None:
-        super().__init__(message)
+    response: Response
+
+    def __init__(self, message: Response) -> None:
+        self.response = message
```
Comment (callumforrester) [inline src/blueapi/cli/amq.py:14]: This de-generliases the exception to be dependant on HTTP. Could make it optional, or make a new, specific exception.

File: src/blueapi/cli/cli.py [hunk 2]
```diff
@@ -158,7 +158,14 @@ def store_finished_event(event: WorkerEvent) -> None:
     parameters = parameters or "{}"
     task = RunPlan(name=name, params=json.loads(parameters))
 
-    resp = client.create_task(task)
+    try:
+        resp = client.create_task(task)
+    except BlueskyRemoteError as e:
+        if e.response.status_code == 404:
+            print(f'Plan "{name}" not found')
```
Comment (callumforrester) [inline src/blueapi/cli/cli.py:165]: Should the server-side API provide a human-readable message in this case?

File: src/blueapi/cli/amq.py (hunk 1 above)
Comment (stan-dot) [reply src/blueapi/cli/amq.py:14]: bluesky remote is only http rest at the moment, right?

we could generalize when we need to

**Comment by stan-dot** [comment]
```     try:
        task = Task(name=name, params=parsed_params)
        resp = client.create_task(task)
        task_id = resp.task_id
    except ValidationError as e:
        pprint(f"failed to validate the task parameters, {task_id}, error: {e}")
        return
    except BlueskyRemoteError as e:
        pprint(f"server error with this message: {e}")
        return
    except ValueError:
        pprint("task could not run")
        return
        
        ```
        
        arguably this exists therefore we can close this PR

**Comment by callumforrester** [comment]
> arguably this exists therefore we can close this PR

Agreed

**Pull request closed without merging**